Source/Plugins/PluginTIFF.cpp

```diff
@@ -2013,12 +2024,16 @@ Load(FreeImageIO *io, fi_handle handle, int page, int flags, void *data) {
 				const uint32_t tileRowSize = (uint32_t)TIFFTileRowSize(tif);
 				const uint32_t imageRowSize = (uint32_t)TIFFScanlineSize(tif);
 
+				if (width / tileWidth * tileRowSize * 8 > bitspersample * samplesperpixel * width) {
```

Comment:
I think more correct condition is `if (width / tileWidth * tileRowSize > dst_pitch)`